stdlib/messaging/rabbitmq/src/main/java/org/ballerinalang/messaging/rabbitmq/util/ConnectionUtils.java

```diff
@@ -83,6 +108,84 @@ public static Connection createConnection(MapValue<String, Object> connectionCon
         }
     }
 
+    /**
+     * Creates and retrieves the initialized SSLContext.
+     *
+     * @param secureSocket secureSocket record.
+     * @return Initialized SSLContext.
+     */
+    private static SSLContext getSSLContext(MapValue secureSocket) {
+        final String CERT_PASS = "password";
+        final String CERT_PATH = "path";
+        try {
+            MapValue cryptoKeyStore = secureSocket.getMapValue(RabbitMQConstants.RABBITMQ_CONNECTION_KEYSTORE);
+            MapValue cryptoTrustStore = secureSocket.getMapValue(RabbitMQConstants.RABBITMQ_CONNECTION_TRUSTORE);
+
+            char[] keyPassphrase = cryptoKeyStore.getStringValue(CERT_PASS).toCharArray();
+            String keyFilePath = cryptoKeyStore.getStringValue(CERT_PATH);
+
+            char[] trustPassphrase = cryptoTrustStore.getStringValue(CERT_PASS).toCharArray();
+            String trustFilePath = cryptoTrustStore.getStringValue(CERT_PATH);
+
+            String tlsVersion = secureSocket.getStringValue(RabbitMQConstants.RABBITMQ_CONNECTION_TLS_VERSION);
+            if (tlsVersion == null) {
+                tlsVersion = "TLS";
+            }
+
+            KeyStore keyStore = KeyStore.getInstance(RabbitMQConstants.KEY_STORE_TYPE);
+            if (keyFilePath != null) {
+                keyStore.load(new FileInputStream(keyFilePath), keyPassphrase);
```

Comment:
We need to close the FileInputStream once we done use it.

Reply:
Resolved